src/hash_to_curve/map_g1.rs

```diff
@@ -0,0 +1,961 @@
+        0x189e_5fe4_470c_d73c,
+    ]),
+    Fp::from_raw_unchecked([
+        0x7ebd_546c_a157_5ed2,
+        0xe47d_5a98_1d08_1b55,
+        0x57b2_b625_b6d4_ca21,
+        0xb0a1_ba04_2285_20cc,
+        0x9873_8983_c210_7ff3,
+        0x13dd_dbc4_799d_81d6,
+    ]),
+    Fp::from_raw_unchecked([
+        0x0931_9f2e_3983_4935,
+        0x039e_952c_bdb0_5c21,
+        0x55ba_77a9_a2f7_6493,
+        0xfd04_e3df_c608_6467,
+        0xfb95_832e_7d78_742e,
+        0x0ef9_c24e_ccaf_5e0e,
+    ]),
+    Fp::from_raw_unchecked([
+        0x7609_0000_0002_fffd,
+        0xebf4_000b_c40c_0002,
+        0x5f48_9857_53c7_58ba,
+        0x77ce_5853_7052_5745,
+        0x5c07_1a97_a256_ec6d,
+        0x15f6_5ec3_fa80_e493,
+    ]),
+];
+
+const SSWU_ELLP_A: Fp = Fp::from_raw_unchecked([
+    0x2f65_aa0e_9af5_aa51,
+    0x8646_4c2d_1e84_16c3,
+    0xb85c_e591_b7bd_31e2,
+    0x27e1_1c91_b5f2_4e7c,
+    0x2837_6eda_6bfc_1835,
+    0x1554_55c3_e507_1d85,
+]);
+
+const SSWU_ELLP_B: Fp = Fp::from_raw_unchecked([
+    0xfb99_6971_fe22_a1e0,
+    0x9aa9_3eb3_5b74_2d6f,
+    0x8c47_6013_de99_c5c4,
+    0x873e_27c3_a221_e571,
+    0xca72_b5e4_5a52_d888,
+    0x0682_4061_418a_386b,
+]);
+
+const SSWU_XI: Fp = Fp::from_raw_unchecked([
+    0x886c_0000_0023_ffdc,
+    0x0f70_008d_3090_001d,
+    0x7767_2417_ed58_28c3,
+    0x9dac_23e9_43dc_1740,
+    0x5055_3f1b_9c13_1521,
+    0x078c_712f_be0a_b6e8,
+]);
+
+const SQRT_M_XI_CUBED: Fp = Fp::from_raw_unchecked([
+    0x43b5_71ca_d321_5f1f,
+    0xccb4_60ef_1c70_2dc2,
+    0x742d_884f_4f97_100b,
+    0xdb2c_3e32_38a3_382b,
+    0xe40f_3fa1_3fce_8f88,
+    0x0073_a2af_9892_a2ff,
+]);
+
+impl HashToField for Fp {
+    // ceil(log2(p)) = 381, m = 1, k = 128.
+    type InputLength = U64;
+
+    fn from_okm(okm: &GenericArray<u8, U64>) -> Fp {
+        const F_2_256: Fp = Fp::from_raw_unchecked([
+            0x075b_3cd7_c5ce_820f,
+            0x3ec6_ba62_1c3e_db0b,
+            0x168a_13d8_2bff_6bce,
+            0x8766_3c4b_f8c4_49d2,
+            0x15f3_4c83_ddc8_d830,
+            0x0f96_28b4_9caa_2e85,
+        ]);
+
+        let mut bs = [0u8; 48];
+        bs[16..].copy_from_slice(&okm[..32]);
+        let db = Fp::from_bytes(&bs).unwrap();
+
+        bs[16..].copy_from_slice(&okm[32..]);
+        let da = Fp::from_bytes(&bs).unwrap();
+
+        db * F_2_256 + da
+    }
+}
+
+impl Sgn0 for Fp {
+    fn sgn0(&self) -> Choice {
+        // Turn into canonical form by computing
+        // (a.R) / R = a
+        let tmp = Fp::montgomery_reduce(
+            self.0[0], self.0[1], self.0[2], self.0[3], self.0[4], self.0[5], 0, 0, 0, 0, 0, 0,
+        );
+        Choice::from((tmp.0[0] & 1) as u8)
+    }
+}
+
+/// Maps an element of [`Fp`] to a point on iso-G1.
+///
+/// Implements [section 6.6.2 of `draft-irtf-cfrg-hash-to-curve-11`][sswu].
+///
+/// [sswu]: https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-hash-to-curve-11#section-6.6.2
+fn map_to_curve_simple_ssw(u: &Fp) -> G1Projective {
```

Comment:
```suggestion
fn map_to_curve_simple_swu(u: &Fp) -> G1Projective {
```
(and similarly below)

Reply:
I think I fixed this already?

Reply:
No, it got truncated and then that was reverted, but the typo is still there AFAICT.